haystack/components/joiners/answer_joiner.py

```diff
@@ -0,0 +1,169 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import itertools
+from enum import Enum
+from math import inf
+from typing import Any, Callable, Dict, List, Optional, Union
+
+from haystack import component, default_from_dict, default_to_dict, logging
+from haystack.core.component.types import Variadic
+from haystack.dataclasses.answer import ExtractedAnswer, ExtractedTableAnswer, GeneratedAnswer
+
+AnswerType = Union[GeneratedAnswer, ExtractedTableAnswer, ExtractedAnswer]
+
+logger = logging.getLogger(__name__)
+
+
+class JoinMode(Enum):
+    """
+    Enum for AnswerJoiner join modes.
+    """
+
+    CONCATENATE = "concatenate"
+
+    def __str__(self):
+        return self.value
+
+    @staticmethod
+    def from_str(string: str) -> "JoinMode":
+        """
+        Convert a string to a JoinMode enum.
+        """
+        enum_map = {e.value: e for e in JoinMode}
+        mode = enum_map.get(string)
+        if mode is None:
+            msg = f"Unknown join mode '{string}'. Supported modes in AnswerJoiner are: {list(enum_map.keys())}"
+            raise ValueError(msg)
+        return mode
+
+
+@component
+class AnswerJoiner:
+    """
+    The `AnswerJoiner` is useful for merging of multiple lists of `Answer` objects into a single unified list.
+
+    This component is useful in scenarios where you have answers from different generators and need to combine
+    them into a single list. One option of merging is to use predefined join modes, such as `CONCATENATE`, which
+    keeps the highest scored answer in case of duplicates. Another option is to provide a custom join function
+    that takes a list of lists of `Answer` objects and returns a single list of `Answer` objects.
```

Comment:
Ahh good point, I'll update the release notes. Opted to remove the custom join function for now since there some problems in the implementation and I don't think there is a strong need for it right now.